skrub/_datetime_encoder.py

```diff
@@ -18,78 +22,456 @@
     "microsecond": "us",
     "nanosecond": "N",
 }
-TIME_LEVELS: list[str] = list(WORD_TO_ALIAS.keys())
-AcceptedTimeValues = Literal[
-    "year",
-    "month",
-    "day",
-    "hour",
-    "minute",
-    "second",
-    "microsecond",
-    "nanosecond",
-]
-
-
-class DatetimeEncoder(BaseEstimator, TransformerMixin):
-    """Transform each datetime column into several numeric columns \
-    for temporal features (e.g. "year", "month", "day"...).
-
-    Constant extracted features are dropped; for instance, if the year is
-    always the same in a feature, the extracted "year" column won't be added.
+TIME_LEVELS = list(WORD_TO_ALIAS)
+
+
+def _is_pandas_format_mixed_available():
+    pandas_version = pd.__version__
+    min_pandas_version = "2.0.0"
+    return parse_version(min_pandas_version) < parse_version(pandas_version)
+
+
+MIXED_FORMAT = "mixed" if _is_pandas_format_mixed_available() else None
+
+
+def to_datetime(
+    X,
+    errors="coerce",
+    **kwargs,
+):
+    """
+    Convert argument to datetime. Return the input if not datetime-parsable.
+
+    Augment :func:`pandas.to_datetime` by supporting dataframes
+    and 2d arrays inputs. It converts compatible columns to datetime, and
+    pass incompatible columns unchanged.
+
+    int, float, str, datetime, list, tuple, 1d array, and Series are defered to
+    :func:`pandas.to_datetime` directly.
+
+    Parameters
+    ----------
+    X : int, float, str, datetime, list, tuple, nd array, Series, DataFrame/dict-like
+        The object to convert to a datetime.
+
+    errors : {'ignore', 'raise', 'coerce'}, default 'coerce'
+        - If ``'raise'``, then invalid parsing will raise an exception.
+        - If ``'coerce'``, then invalid parsing will be set as ``NaT``.
+        Note that ``'ignore'`` is not used and will raise an error.
+
+    **kwargs : key, value mappings
+        Other keyword arguments are passed down to :func:`pandas.to_datetime`.
+        Raise an error if 'unit' is set to any value. This is because, in
+        `pandas.to_datetime`, unit is specific to timestamps, whereas in
+        `skru`.to_datetime` we don't attempt to parse numeric columns.
+
+    Returns
+    -------
+    datetime
+        Return type depends on input.
+        - dataframes, series and 2d arrays return the same type
+        - otherwise return the same output as :func:`pandas.to_datetime`.
+
+    See Also
+    --------
+    :func:`pandas.to_datetime`
+        Convert argument to datetime.
+
+    Examples
+    --------
+    >>> X = pd.DataFrame(dict(a=[1, 2], b=["2021-01-01", "2021-02-02"]))
+    >>> X = to_datetime(X)
+    >>> X.dtypes.to_list()
+    [dtype('int64'), dtype('<M8[ns]')]
+    """
+    errors_options = ["coerce", "raise"]
+    if errors not in errors_options:
+        raise ValueError(f"errors options are {errors_options!r}, got {errors!r}.")
+    kwargs["errors"] = errors
+
+    if kwargs.get("unit", None) is not None:
+        raise ValueError("unit ")
```

Comment:
```suggestion
        raise ValueError("'unit' is not a parameter of skrub.to_datetime; it is only meaningful when applying pandas.to_datetime to a numerical column")
```

Reply:
haha it seems that something is off with my last commit